library/hawkular_alerts_group_dampening.py

```diff
@@ -0,0 +1,304 @@
+
+    def list_group_dampenings(self, group_id):
+        """ Lists all dampenings in the group trigger
+
+        Returns:
+            whether or not a change took place and a short message
+            describing the operation executed
+        """
+        if not self.group_trigger_exist(group_id):
+            self.module.fail_json(msg="Group trigger {group_id} doesn't exist".format(group_id=group_id))
+        group_dampenings = self.get_group_dampenings(group_id)
+        return dict(
+            msg="Successfully listed group dampenings",
+            changed=self.changed,
+            group_dampenings=group_dampenings)
+
+    def update_required(self, desired_dampening, current_dampening):
+        """ Checks if an update is required to the current dampening
+
+            Returns:
+               True, if an update is required for the dampening, False otherwise
+        """
+        for key in desired_dampening.keys():
+            if desired_dampening[key] != current_dampening.get(key):
+                return True
+        return False
+
+    def update_group_dampening(self, group_id, dampening_id, dampening):
+        """ Updates a group trigger dampening in Hawkular Alerts
+
+            Returns:
+                whether or not a change took place and a short message
+                describing the operation executed
+        """
+        dampening_object = hawkular.alerts.Dampening(dampening)
+        try:
+            self.client.update_group_dampening(group_id, dampening_id, dampening_object)
+        except Exception as e:
+            self.module.fail_json(msg="Failed to update dampening. Error: {error}".format(error=e))
+        self.changed = True
+
+    def create_group_dampening(self, group_id, dampening):
+        """ Creates a group trigger dampening in Hawkular Alerts
+
+            Returns:
+                whether or not a change took place and a short message
+                describing the operation executed
+        """
+        new_dampening = hawkular.alerts.Dampening(dampening)
+        try:
+            self.client.create_group_dampening(group_id, new_dampening)
+        except Exception as e:
+            self.module.fail_json(msg="Failed to create dampening. Error: {error}".format(error=e))
+        self.changed = True
+
+    def create_or_update_group_dampenings(self, group_id, dampenings):
+        """ Creates or updates group trigger dampening in Hawkular Alerts
+
+            Returns:
+                whether or not a change took place and a short message
+                describing the operation executed
+        """
+        if not self.group_trigger_exist(group_id):
+            self.module.fail(msg="Group trigger (group_id} does not exist ".format(group_id=group_id))
+        messages = []
+        current_dampenings = self.get_group_dampenings(group_id)
+        current_dampenings_by_trigger_mode = {d["trigger_mode"]: d for d in current_dampenings}
+        for dampening in dampenings:
```

Comment:
here `dampening` variable the trigger mode of a desired dampening, right?
IMO confusing, I'd expect `dampening` to be a Dampening.

Using k, v loops may make it more readable what data you're working on:
```
for trigger_mode, desired_dampening in dampenings.items():
    if trigger_mode in current_dampenings_by_trigger_mode:
        current_dampening = current_dampenings_by_trigger_mode[trigger_mode]
        dampening_id = current_dampening["dampening_id"]
        if self.update_required(desired_dampening, current_dampening):
            self.update_group_dampening(group_id, dampening_id, desired_dampening)
        ...
    else:
        desired_dampening["trigger_mode"] = trigger_mode
        self.create_group_dampening(group_id, desired_dampening)
        ...
```